### PR TITLE
HIVE-25919: ClassCastException when pushing boolean column predicate in HBaseStorageHandler

### PR DESCRIPTION
**Pull request by zabetak**
### What changes were proposed in this pull request?

The predicate analyzer can only handle expressions of the form:
1. `col comp_op constant`
2. `constant comp_op col`
and these are always `ExprNodeGenericFuncDesc`.

The new if clause ensures that we are never going to handle anything that is not supported.

### Why are the changes needed?
Avoids the `ClassCastException` during predicate pushdown.

### Does this PR introduce _any_ user-facing change?
No apart from solving the problem.

### How was this patch tested?
`mvn test -Dtest=TestHBaseCliDriver -Dqfile="hbase_ppd_boolean_cols.q"`

### CODE REVIEW

File: hbase-handler/src/test/queries/positive/hbase_ppd_boolean_cols.q [hunk 1]
```diff
@@ -0,0 +1,8 @@
+CREATE TABLE hbase_table(row_key string, c1 boolean, c2 boolean)
+STORED BY 'org.apache.hadoop.hive.hbase.HBaseStorageHandler'
+WITH SERDEPROPERTIES (
+"hbase.columns.mapping" = ":key,cf:c1,cf:c2"
+);
+
```
Comment (abstractdog) [inline hbase-handler/src/test/queries/positive/hbase_ppd_boolean_cols.q:6]: even if this is about a compile-time issue, can include a real query too, not just an explain?

**Review by abstractdog — CHANGES_REQUESTED** [review]
LGTM, only a minor comment

File: hbase-handler/src/test/queries/positive/hbase_ppd_boolean_cols.q (hunk 1 above)
Comment (zabetak) [reply hbase-handler/src/test/queries/positive/hbase_ppd_boolean_cols.q:13]: Fixed in https://github.com/apache/hive/pull/2996/commits/42333553fbac3949e4f0c65e21b6237ca171b971. Please have another look when you find some time.

**Review by abstractdog — APPROVED** [review]

**Comment by abstractdog** [comment]
+1, very neat, @zabetak

**Pull request closed without merging**